src/App.test.js

```diff
@@ -1,10 +1,36 @@
 import React from "react";
 import { shallow, mount } from "enzyme";
 import App from "./App";
+
+let app;
+let appInstance;
+let originalState;
+
+beforeEach(() => {
+     app = shallow(<App />);
+     appInstance = app.instance();
+     originalState = JSON.parse(JSON.stringify(app.instance().state));
```

Comment:
Stringify and the parse? Don't they basically cancel each other out? Why do we need to do this?

Reply:
This is done to deep copy the state object rather than assigning the state object to originalState.

Reply:
Actually, consider using [immutable.js](https://github.com/facebook/immutable-js) here. Since React recommends immutability of state, moving to immutability paradigm right now might help in the long term. 

With immutable.js, making a copy is as easy as just making another reference.

Wrote this to illustrate the point. 
```javascript
const { Map } = require("immutable");

it("test i", () => {immutability
  const map = Map({ a: 1, b: 2, c: 3 });
  const aCopyOfMap_NotJustAnotherReference = map;

  map.set("a", 200);

  console.log(aCopyOfMap_NotJustAnotherReference); // This will still print { a: 1, b: 2, c: 3 }

});
```

Reply:
BTW, we are already taking a dependency on Immutable in this pull request. https://github.com/s-maheshbabu/silent-alexa/pull/9/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2

Reply:
I will not make this change in this pull request as it will pollute the current CR. There are other tests that require this change and they can be bundled together.

Reply:
In that case, open an issue on Github and put the link here. Assign the issue to yourself.

Reply:
Created an issue to address this: https://github.com/s-maheshbabu/silent-alexa/issues/11